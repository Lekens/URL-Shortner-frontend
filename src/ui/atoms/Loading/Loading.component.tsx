import styles from './styles.module.scss';

function Loading() {
  return (
    <div className={styles.wrapper}>
      <div className={styles.loader}>Loading...</div>
    </div>
  );
}
export default Loading;
