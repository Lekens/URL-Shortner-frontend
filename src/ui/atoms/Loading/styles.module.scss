@import 'features/common/assets/styles/variables.module.scss';

.wrapper {
  padding: 20px 0;
}

.loader {
  position: relative;
  width: 11em;
  height: 11em;
  margin: 50px auto;
  animation: load3 1.4s infinite linear;
  background: $white;
  background: linear-gradient(to right, #fff 10%, rgba(255, 255, 255, 0) 42%);
  border-radius: 50%;
  font-size: 10px;
  text-indent: -9999em;
  transform: translateZ(0);
}

.loader::before {
  position: absolute;
  top: 0;
  left: 0;
  width: 50%;
  height: 50%;
  background: $white;
  border-radius: 100% 0 0 0;
  content: '';
}

.loader::after {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  width: 75%;
  height: 75%;
  margin: auto;
  background: $gray-black;
  border-radius: 50%;
  content: '';
}

@keyframes load3 {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
