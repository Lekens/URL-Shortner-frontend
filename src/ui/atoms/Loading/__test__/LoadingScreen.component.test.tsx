import React from 'react';

import { render, screen } from '@testing-library/react';
import Loading from 'ui/atoms/Loading/Loading.component';

describe('LoadingScreen component', () => {
  it('should match snapshot', () => {
    const { container } = render(<Loading />);
    expect(container).toMatchSnapshot();
    expect(screen.queryByText('Loading...')).toBeInTheDocument();
  });
});
