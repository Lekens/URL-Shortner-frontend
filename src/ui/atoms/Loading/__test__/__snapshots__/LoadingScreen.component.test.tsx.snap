// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingScreen component should match snapshot 1`] = `
<div>
  <div
    class="wrapper"
  >
    <div
      class="loader"
    >
      Loading...
    </div>
  </div>
</div>
`;
